_ml-commons-plugin/agents-tools/index.md

```diff
@@ -0,0 +1,128 @@
+## Agents
+
+An _agent_ is a coordinator that uses a large language model (LLM) to solve a problem. After the LLM reasons and decides what action to take, the agent coordinates the action execution. OpenSearch supports the following agent types:
+
+- [_Flow agent_](#flow-agents): Runs tools sequentially, in the order specified in its configuration. The workflow of a flow agent is fixed. Useful for retrieval-augmented generation (RAG).
+- [_Conversational agent_](#conversational-agents): Reasons to provide a response based on the available knowledge. The workflow of a conversational agent is variable, based on follow-up questions. For specific questions, uses the Chain-of-Thought (CoT) process to select the best tool for providing a response to the question out of the configured tools. Useful for chatbot creation.
+
+### Flow agents
+
+A flow agent is configured with a set of tools that it runs in order. For example, the following agent runs the `VectorDBTool` and then the `MLModelTool`. The agent coordinates the tools so one tool's output can become another tool's input. In this example, the `VectorDBTool` queries the k-NN index and the agent passes its output `${parameters.VectorDBTool.output}` to the `MLModelTool` as a context, along with the `${parameters.question}` (see the `prompt` parameter):
+
+```json
+POST /_plugins/_ml/agents/_register
+{
+  "name": "Test_Agent_For_RAG",
+  "type": "flow",
+  "description": "this is a test agent",
+  "tools": [
+    {
+      "type": "VectorDBTool",
+      "parameters": {
+        "model_id": "aVeif4oB5Vm0Tdw8zYO2",
+        "index": "my_test_data",
+        "embedding_field": "embedding",
+        "source_field": ["text"],
+        "input": "${parameters.question}"
+      }
+    },
+    {
+      "type": "MLModelTool",
+      "description": "A general tool to answer any question",
+      "parameters": {
+        "model_id": "NWR9YIsBUysqmzBdifVJ",
+        "prompt": "\n\nHuman:You are a professional data analyst. You will always answer a question based on the given context first. If the answer is not directly shown in the context, you will analyze the data and find the answer. If you don't know the answer, just say you don't know. \n\n Context:\n${parameters.VectorDBTool.output}\n\nHuman:${parameters.question}\n\nAssistant:"
+      }
+    }
+  ]
+}
+```
+
+### Conversational agents
+
+A conversational agent can be configured with an LLM and a set of supplementary tools that perform specific jobs. For example, you can set up an LLM and a `CATIndexTool` when configuring an agent. When you send a question to the model, the agent also includes the `CATIndexTool` as context. The LLM then decides whether it needs to use a the `CATIndexTool` to answer questions like "How many indexes are in my cluster?" The context allows an LLM to answer specific questions that are outside of its knowledge base. For example, the following agent is configured with an LLM and a `CATIndexTool` that calls the `_cat/indices` API:
```

Comment:
Actually the catindextool doesn't call rest api _cat/indices directly, but this tool does the same thing in another way through transport api.